statsmodels/tsa/filters/hp_filter.py

```diff
@@ -98,4 +108,13 @@ def hpfilter(x, lamb=1600):
     trend = spsolve(I+lamb*K.T.dot(K), x, use_umfpack=use_umfpack)
 
     cycle = x - trend
+    if not two_sided:
+        one_sided_cycle = [0, 0]
+        one_sided_trend = list(x[:2])
+        for t in range(3, nobs):
+            cycle_t, trend_t = hpfilter(x[:t], lamb=lamb, two_sided=True)
```

Comment:
Is this efficient?  Surely there is a better 1-sided than running `nobs` hp filters.

Reply:
Looks like it should be implemented as a state-space model

https://stats.stackexchange.com/questions/48326/state-space-formulation-of-hodrick-prescott-%ef%ac%81lter/48336#48336